src/memtable.rs

```diff
@@ -200,8 +218,50 @@ impl<E: Message + Clone, W: EntryExt<E>> MemTable<E, W> {
         }
     }
 
+    pub fn rewrite(&mut self, entries_index: Vec<EntryIndex>, latest_rewrite: u64) {
+        if entries_index.is_empty() || self.entries_index.is_empty() {
+            return;
+        }
+
+        let first = entries_index.first().unwrap().index;
+        let front = self.entries_index[self.rewrite_count].index;
+        assert!(front >= first);
```

Comment:
How about add `assert_eq!(first, self.rewrite_count);`?

Reply:
What if `self.entries_index` has been changed so that  the first index is not equal to the first index of `self.entries_index`?

Reply:
So `first` could not equal to `rewrite_count`. It's considered in the later code.